exporter/metric/metric.go

```diff
@@ -276,21 +319,80 @@ func (me *metricExporter) recordToMdpb(record *export.Record) *googlemetricpb.Me
 // proto type for Cloud Monitoring.
 //
 // https://cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.monitoredResourceDescriptors
-func (me *metricExporter) resourceToMonitoredResourcepb(_ *resource.Resource) *monitoredrespb.MonitoredResource {
-	// TODO: Implement the process to convert Resource to MonitoringResoruce.
-	// ref. https://cloud.google.com/monitoring/api/resources
-	// This method only returns global resource for now.
+func (me *metricExporter) resourceToMonitoredResourcepb(res *resource.Resource) *monitoredrespb.MonitoredResource {
+
+	if res == nil || res.Len() == 0 {
+    	// Return "global" Monitored resources if the input resource is null or empty
+	    // "global" only accepts "project_id" for label.
+	    // https://cloud.google.com/monitoring/api/resources#tag_global
+		return &monitoredrespb.MonitoredResource{
+			Type: "global",
+			Labels: map[string]string{
+				"project_id": me.o.ProjectID,
+			},
+		}	
+	}
+
+	// labels in Resource are in the form of []kv.KeyValue
+	// convert them into a map of kv.String
+	resLabelList := res.Attributes()
+
+	resLabelMap := make(map[string]string)
+	for _, label := range resLabelList {
+		resLabelMap[string(label.Key)] = label.Value.AsString()
+	}
+
+	resTypeStr := "global"
+	monitoredReslabelsMap := make(map[string]string)
+	match := genericResourceMap
+	
+	if resType, found := resLabelMap["type"]; found {
```

Comment:
I assume this PR is still work in progress, but the overall approach looks like its in the right direction.

Note though that OpenTelemetry resources no longer have a "type" property, but instead just a list of attributes where the prefix of the attribute can be used to determine the type (e.g. the k8s prefix in the key "**k8s**.pod.name"). Check the implementation in the JS operations exporter for reference as to how they've done this mapping.

Reply:
Thanks for pointing out. The JS version used CloudProvider to differentiate the type. (https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/resource/semantic_conventions#cloud) However, there are only three types: "aws", "gcp", and "azure". Since we need "k8s", I guess we need to talk with backend team to add a new constant "k8s"?

Reply:
I think "k8s" is a deployment service (https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/resource/semantic_conventions#kubernetes) and is not related to the environment (https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/resource/semantic_conventions#environment). The environment probably requires a different resource type other than "k8s"

Reply:
You're right that the Javascript implementaton is incomplete. However, the approach you're currently taking to determine if the resource type is incorrect.

As per Duke's comment, "K8s" is a service rather than a Cloud Provider, so will never show up as a possible value for "cloud.provider". To determine if we are running on a K8s Container, you should be able to just do something like checking for the existance of a `k8s.*` attribute (e.g. `k8s.pod.name`?).

If we wanted to only record info for K8s Containers running on GKE, we could additionally check `cloud.provider == "gcp"`. Note that would likely also include K8s clusters running on GCE directly, I'm not sure if we would want to distinguish those cases but if we did, we'd likely need to establish a new semantic attribute such as `k8s.provider.name`